### PR TITLE
update compiler version and remove deprecated entities

### PR DESCRIPTION
**Pull request by nichonien**
This PR contains changes : 

1. Upgrade compiler version
2. Replaces `now` with `block.timestamp`
3. Replaces `byte` with `bytes1`
4. Updates `keccak256` implementation to one parameter implementation (latest version)
5. Updates `0x0` address to full length address

### CODE REVIEW

**Comment by nichonien** [comment]
@mirceanis Did you get a chance to look the at PR?

File: contracts/EthereumDIDRegistry.sol [hunk 1]
```diff
@@ -65,13 +65,13 @@ contract EthereumDIDRegistry {
   }
 
   function changeOwnerSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, address newOwner) public {
-    bytes32 hash = keccak256(byte(0x19), byte(0), this, nonce[identityOwner(identity)], identity, "changeOwner", newOwner);
+    bytes32 hash = keccak256(abi.encode(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "changeOwner", newOwner));
```
Comment (mirceanis) [inline contracts/EthereumDIDRegistry.sol:68]: To preserve compatibility with existing codebases, this should be `encodePacked`
```suggestion
    bytes32 hash = keccak256(abi.encodePacked(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "changeOwner", newOwner));
```

File: contracts/EthereumDIDRegistry.sol [hunk 2]
```diff
@@ -80,13 +80,13 @@ contract EthereumDIDRegistry {
   }
 
   function addDelegateSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, bytes32 delegateType, address delegate, uint validity) public {
-    bytes32 hash = keccak256(byte(0x19), byte(0), this, nonce[identityOwner(identity)], identity, "addDelegate", delegateType, delegate, validity);
+    bytes32 hash = keccak256(abi.encode(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "addDelegate", delegateType, delegate, validity));
```
Comment (mirceanis) [inline contracts/EthereumDIDRegistry.sol:83]: To preserve compatibility with existing codebases, this should be `encodePacked`
```suggestion
    bytes32 hash = keccak256(abi.encodePacked(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "addDelegate", delegateType, delegate, validity));
```

File: contracts/EthereumDIDRegistry.sol [hunk 3]
```diff
@@ -95,36 +95,36 @@ contract EthereumDIDRegistry {
   }
 
   function revokeDelegateSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, bytes32 delegateType, address delegate) public {
-    bytes32 hash = keccak256(byte(0x19), byte(0), this, nonce[identityOwner(identity)], identity, "revokeDelegate", delegateType, delegate);
+    bytes32 hash = keccak256(abi.encode(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "revokeDelegate", delegateType, delegate));
```
Comment (mirceanis) [inline contracts/EthereumDIDRegistry.sol:98]: To preserve compatibility with existing codebases, this should be `encodePacked`
```suggestion
    bytes32 hash = keccak256(abi.encodePacked(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "revokeDelegate", delegateType, delegate));
```

File: contracts/EthereumDIDRegistry.sol [hunk 4]
```diff
@@ -95,36 +95,36 @@ contract EthereumDIDRegistry {
...
 
-  function setAttribute(address identity, bytes32 name, bytes value, uint validity) public {
+  function setAttribute(address identity, bytes32 name, bytes memory value, uint validity) public {
     setAttribute(identity, msg.sender, name, value, validity);
   }
 
-  function setAttributeSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, bytes32 name, bytes value, uint validity) public {
-    bytes32 hash = keccak256(byte(0x19), byte(0), this, nonce[identityOwner(identity)], identity, "setAttribute", name, value, validity);
+  function setAttributeSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, bytes32 name, bytes memory value, uint validity) public {
+    bytes32 hash = keccak256(abi.encode(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "setAttribute", name, value, validity));
```
Comment (mirceanis) [inline contracts/EthereumDIDRegistry.sol:112]: To preserve compatibility with existing codebases, this should be `encodePacked`
```suggestion
    bytes32 hash = keccak256(abi.encodePacked(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "setAttribute", name, value, validity));
```

File: contracts/EthereumDIDRegistry.sol (hunk 3 above)
Comment (mirceanis) [inline contracts/EthereumDIDRegistry.sol:98]: To preserve compatibility with existing codebases, this should be `encodePacked`
```suggestion
    bytes32 hash = keccak256(abi.encodePacked(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "revokeDelegate", delegateType, delegate));
```

File: contracts/EthereumDIDRegistry.sol [hunk 5]
```diff
@@ -95,36 +95,36 @@ contract EthereumDIDRegistry {
...
 
-  function revokeAttribute(address identity, bytes32 name, bytes value) public {
+  function revokeAttribute(address identity, bytes32 name, bytes memory value) public {
     revokeAttribute(identity, msg.sender, name, value);
   }
 
- function revokeAttributeSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, bytes32 name, bytes value) public {
-    bytes32 hash = keccak256(byte(0x19), byte(0), this, nonce[identityOwner(identity)], identity, "revokeAttribute", name, value); 
+ function revokeAttributeSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, bytes32 name, bytes memory value) public {
+    bytes32 hash = keccak256(abi.encode(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "revokeAttribute", name, value));
```
Comment (mirceanis) [inline contracts/EthereumDIDRegistry.sol:126]: To preserve compatibility with existing codebases, this should be `encodePacked`
```suggestion
    bytes32 hash = keccak256(abi.encodePacked(bytes1(0x19), bytes1(0), this, nonce[identityOwner(identity)], identity, "revokeAttribute", name, value)); 
```

**Review by mirceanis — CHANGES_REQUESTED** [review]
Excellent work.

It took me a while to test this out since the existing test harness doesn't work on my machine any more.
We need to upgrade the test automation here as well if we want to keep improving the contract(s).

I made some suggestions regarding the way the message is hashed in the `change/add/remove*Signed()` methods, for maintaining compatibility with existing codebases.

Using `abi.encodePacked` allows the caller to use simple concatenation of byte arrays instead of doing RLP encoding of arguments.
Also, since there is at most one variable-length array parameter, the [warning in the solidity docs](https://docs.soliditylang.org/en/v0.8.10/abi-spec.html#non-standard-packed-mode) does not apply.

**Review by mirceanis — APPROVED** [review]
Looks great!

Thank you for contributing

**Comment by mirceanis** [comment]
BTW, I have started a governance topic about did:ethr upgrades on our discord: https://discord.gg/MTeTAwSYe7

Feel free to join and share your thoughts.

**Pull request merged**